### ISSUE TITLE
Benchmarks for binary vectors?

### ISSUE THREAD

**Issue by thvasilo** [erikbern/ann-benchmarks#162]
[FAISS](https://github.com/facebookresearch/faiss/wiki/Binary-indexes) and I assume some other libraries include optimized algorithms for binary features (both exact like [Multi-Index Hashing](http://www.cs.toronto.edu/~norouzi/research/papers/multi_index_hashing.pdf) and approximate).

I was wondering if there are any plans to include such specializations? I assume SIFT features would be good candidates for data.

**Comment by maumueller** [comment]
This already exists with two datasets at http://ann-benchmarks.com/ (Distance: Hamming). In particular, MIH is supported already.

Would be nice to have the FAISS binary variants, though. Feel free to add it via a PR. :-)

**Comment by thvasilo** [comment]
Gave it a shot today, but I think I want to clarify a couple of things. Is it OK to open a WIP PR to ask for help fixing a couple of potential issues?

**Comment by maumueller** [comment]
Sorry for the late answer. Please submit a [WIP] PR, yes.

**Comment by tim-hilt** [comment]
> This already exists with two datasets at http://ann-benchmarks.com/ (Distance: Hamming)

@maumueller i couldn't find it when i looked there today! Were binary benchmarks excluded again?

**Comment by maumueller** [comment]
Yes, I think the website currently reflects only the latest run of @erikbern. We could put it into the results manually, but this will mix up different versions of implementations. :-(

**Comment by erikbern** [comment]
I can include it next time I run the benchmarks!

**Comment by tim-hilt** [comment]
@erikbern that would be great, thanks! :+1:

**Comment by tim-hilt** [comment]
@erikbern Any progress on this?

**Comment by erikbern** [comment]
Sorry, haven't had time to work on it. I hope to have plenty of free time in March

**Comment by tim-hilt** [comment]
Thanks for the feedback @erikbern :) Great to here, that it's not forgotten!

**Comment by Charlyo** [comment]
Any new regarding this? Would appreciate having binary vectors benchmarks :)
